crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -1026,235 +1068,83 @@ impl PgIndexerStore {
-            })
-            .tap_err(|e| {
-                tracing::error!("Failed to persist tx_calls_mod with error: {}", e);
-            })
-        }));
-
-        futures.push(self.spawn_blocking_task(move |this| {
-            let now = Instant::now();
-            let rows_len = funs.len();
-            transactional_blocking_with_retry!(
-                &this.blocking_cp,
-                |conn| {
-                    for chunk in funs.chunks(PG_COMMIT_CHUNK_SIZE_INTRA_DB_TX) {
-                        insert_or_ignore_into!(tx_calls_fun::table, chunk, conn);
-                    }
-                    Ok::<(), IndexerError>(())
-                },
-                PG_DB_COMMIT_SLEEP_DURATION
-            )
-            .tap_ok(|_| {
-                let elapsed = now.elapsed().as_secs_f64();
-                info!(elapsed, "Persisted {} rows to tx_calls_fun table", rows_len);
-            })
-            .tap_err(|e| {
-                tracing::error!("Failed to persist tx_calls_fun with error: {}", e);
-            })
-        }));
-
-        futures.push(self.spawn_blocking_task(move |this| {
-            let now = Instant::now();
-            let calls_len = digests.len();
-            transactional_blocking_with_retry!(
-                &this.blocking_cp,
-                |conn| {
-                    for chunk in digests.chunks(PG_COMMIT_CHUNK_SIZE_INTRA_DB_TX) {
-                        insert_or_ignore_into!(tx_digests::table, chunk, conn);
-                    }
-                    Ok::<(), IndexerError>(())
-                },
-                Duration::from_secs(60)
-            )
-            .tap_ok(|_| {
-                let elapsed = now.elapsed().as_secs_f64();
-                info!(elapsed, "Persisted {} rows to tx_digests tables", calls_len);
-            })
-            .tap_err(|e| {
-                tracing::error!("Failed to persist tx_digests with error: {}", e);
-            })
-        }));
-
-        futures.push(self.spawn_blocking_task(move |this| {
-            let now = Instant::now();
-            let rows_len = kinds.len();
-            transactional_blocking_with_retry!(
-                &this.blocking_cp,
-                |conn| {
-                    for chunk in kinds.chunks(PG_COMMIT_CHUNK_SIZE_INTRA_DB_TX) {
-                        insert_or_ignore_into!(tx_kinds::table, chunk, conn);
-                    }
-                    Ok::<(), IndexerError>(())
-                },
-                Duration::from_secs(60)
-            )
-            .tap_ok(|_| {
-                let elapsed = now.elapsed().as_secs_f64();
-                info!(elapsed, "Persisted {} rows to tx_kinds tables", rows_len);
-            })
-            .tap_err(|e| {
-                tracing::error!("Failed to persist tx_kinds with error: {}", e);
-            })
-        }));
+        transaction_with_retry(&self.pool, PG_DB_COMMIT_SLEEP_DURATION, |conn| {
```

Comment:
Might be misunderstanding but this is slightly different from the existing implementation right? Previously we have a number of futures that are run concurrently, but now we have a single transaction that sequentially goes through each insert statement. If so, this might affect writer performance?

Reply:
No that's an accurate read of this. I wasn't too worried about the impact to perf here though since how and when we do things in parallel or serially isn't super consistent (that and in the grand scheme of things this is still done in parallel to all the other "updates" that are happening on each group of checkpoints to be written).

I was mostly interested in simplifying things, making them more clear, and then it should be a bit easier to determine how and where we should spend time doing perf optimizations